### ISSUE TITLE
[CI] :x-pack:plugin:ml:qa:native-multi-node-tests:javaRestTest failing

### ISSUE THREAD

**Issue by bpintea** [elastic/elasticsearch#88206]
### CI Link

https://gradle-enterprise.elastic.co/s/kffbtolmvupe6

### Repro line

No logged repro line

### Does it reproduce?

No

### Applicable branches

8.3

### Failure excerpt

This tests failed already a couple other times, all in 8.3:
* https://gradle-enterprise.elastic.co/s/xtdzegdlfi6zs
* https://gradle-enterprise.elastic.co/s/66ihtbtk7gejm

While not positive, I think the relevant failure might be:
```
»    ↓ errors and warnings from /dev/shm/elastic+elasticsearch+8.3+periodic+java-fips-matrix/x-pack/plugin/ml/qa/native-multi-node-tests/build/testclusters/javaRestTest-2/logs/es.out ↓
» [2022-06-29T14:31:50,767][WARN ][stderr                   ] [javaRestTest-2] Jun 29, 2022 2:31:50 PM org.bouncycastle.jsse.provider.ProvTrustManagerFactorySpi getDefaultTrustStore
»   ↑ repeated 4 times ↑
» [2022-06-29T14:31:53,507][WARN ][stderr                   ] [javaRestTest-2] INFO: Initializing with trust store at path: /dev/shm/elastic+elasticsearch+8.3+periodic+java-fips-matrix/x-pack/plugin/ml/qa/native-multi-node-tests/build/testclusters/javaRestTest-2/config/cacerts.bcfks
»   ↑ repeated 5 times ↑
» [2022-06-29T14:31:50,361][ERROR][o.e.x.c.s.SSLService     ] [javaRestTest-2] unsupported ciphers [[TLS_AES_256_GCM_SHA384, TLS_AES_128_GCM_SHA256, TLS_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384]] were requested but cannot be used in this JVM, however there are supported ciphers that will be used [[TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA, TLS_RSA_WITH_AES_256_GCM_SHA384, TLS_RSA_WITH_AES_128_GCM_SHA256, TLS_RSA_WITH_AES_256_CBC_SHA256, TLS_RSA_WITH_AES_128_CBC_SHA256, TLS_RSA_WITH_AES_256_CBC_SHA, TLS_RSA_WITH_AES_128_CBC_SHA]]. If you are trying to use ciphers with a key length greater than 128 bits on an Oracle JVM, you will need to install the unlimited strength JCE policy files.
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:53,507][WARN ][stderr                   ] [javaRestTest-2] Jun 29, 2022 2:31:53 PM org.bouncycastle.jsse.provider.ProvTrustManagerFactorySpi getDefaultTrustStore
» [2022-06-29T14:31:54,217][WARN ][o.e.b.BootstrapChecks    ] [javaRestTest-2] HTTPS is required in order to use the token service; please enable HTTPS using the [xpack.security.http.ssl.enabled] setting or disable the token service using the [xpack.security.authc.token.enabled] setting
» [2022-06-29T14:31:54,230][WARN ][o.e.d.FileBasedSeedHostsProvider] [javaRestTest-2] expected, but did not find, a dynamic hosts list at [/dev/shm/elastic+elasticsearch+8.3+periodic+java-fips-matrix/x-pack/plugin/ml/qa/native-multi-node-tests/build/testclusters/javaRestTest-2/config/unicast_hosts.txt]
» [2022-06-29T14:31:56,920][WARN ][stderr                   ] [javaRestTest-2] Jun 29, 2022 2:31:56 PM org.bouncycastle.jsse.provider.ProvTlsServer notifyAlertRaised
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,920][WARN ][stderr                   ] [javaRestTest-2] INFO: Server raised fatal(2) handshake_failure(40) alert: Failed to process record
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,921][WARN ][stderr                   ] [javaRestTest-2] org.bouncycastle.tls.TlsFatalAlert: handshake_failure(40)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,921][WARN ][stderr                   ] [javaRestTest-2] 	at org.bouncycastle.fips.tls/org.bouncycastle.tls.TlsProtocol.handleAlertWarningMessage(TlsProtocol.java:184)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,921][WARN ][stderr                   ] [javaRestTest-2] 	at org.bouncycastle.fips.tls/org.bouncycastle.tls.TlsServerProtocol.handleAlertWarningMessage(TlsServerProtocol.java:413)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,922][WARN ][stderr                   ] [javaRestTest-2] 	at org.bouncycastle.fips.tls/org.bouncycastle.tls.TlsProtocol.handleAlertMessage(TlsProtocol.java:161)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,922][WARN ][stderr                   ] [javaRestTest-2] 	at org.bouncycastle.fips.tls/org.bouncycastle.tls.TlsProtocol.processAlertQueue(TlsProtocol.java:570)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,922][WARN ][stderr                   ] [javaRestTest-2] 	at org.bouncycastle.fips.tls/org.bouncycastle.tls.TlsProtocol.processRecord(TlsProtocol.java:435)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,923][WARN ][stderr                   ] [javaRestTest-2] 	at org.bouncycastle.fips.tls/org.bouncycastle.tls.RecordStream.readFullRecord(RecordStream.java:184)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,923][WARN ][stderr                   ] [javaRestTest-2] 	at org.bouncycastle.fips.tls/org.bouncycastle.tls.TlsProtocol.safeReadFullRecord(TlsProtocol.java:727)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,923][WARN ][stderr                   ] [javaRestTest-2] 	at org.bouncycastle.fips.tls/org.bouncycastle.tls.TlsProtocol.offerInput(TlsProtocol.java:1059)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,924][WARN ][stderr                   ] [javaRestTest-2] 	at org.bouncycastle.fips.tls/org.bouncycastle.tls.TlsProtocol.offerInput(TlsProtocol.java:1027)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,924][WARN ][stderr                   ] [javaRestTest-2] 	at org.bouncycastle.fips.tls/org.bouncycastle.jsse.provider.ProvSSLEngine.unwrap(ProvSSLEngine.java:445)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,924][WARN ][stderr                   ] [javaRestTest-2] 	at java.base/javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:679)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,924][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.handler@4.1.76.Final/io.netty.handler.ssl.SslHandler$SslEngineType$3.unwrap(SslHandler.java:295)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,925][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.handler@4.1.76.Final/io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1342)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,925][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.handler@4.1.76.Final/io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1235)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,925][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.handler@4.1.76.Final/io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1284)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,926][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.codec@4.1.76.Final/io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:510)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,926][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.codec@4.1.76.Final/io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:449)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,926][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.codec@4.1.76.Final/io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:279)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,928][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.transport@4.1.76.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
»   ↑ repeated 4 times ↑
» [2022-06-29T14:31:56,928][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.transport@4.1.76.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
»   ↑ repeated 4 times ↑
» [2022-06-29T14:31:56,927][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.transport@4.1.76.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,928][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.transport@4.1.76.Final/io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,928][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.transport@4.1.76.Final/io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,929][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.transport@4.1.76.Final/io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,929][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.transport@4.1.76.Final/io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,929][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.transport@4.1.76.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:623)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,929][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.transport@4.1.76.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:586)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,929][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.transport@4.1.76.Final/io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,930][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.common@4.1.76.Final/io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,930][WARN ][stderr                   ] [javaRestTest-2] 	at io.netty.common@4.1.76.Final/io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,930][WARN ][stderr                   ] [javaRestTest-2] 	at java.base/java.lang.Thread.run(Thread.java:833)
»   ↑ repeated 2 times ↑
» [2022-06-29T14:31:56,931][WARN ][o.e.t.TcpTransport       ] [javaRestTest-2] exception caught on transport layer [Netty4TcpChannel{localAddress=/127.0.0.1:35259, remoteAddress=/127.0.0.1:52198, profile=default}], closing connection
»  io.netty.handler.codec.DecoderException: javax.net.ssl.SSLException: org.bouncycastle.tls.TlsFatalAlert: handshake_failure(40)
»  	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:480) ~[netty-codec-4.1.76.Final.jar:?]
»  	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:279) ~[netty-codec-4.1.76.Final.jar:?]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.76.Final.jar:?]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.76.Final.jar:?]
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.76.Final.jar:?]
»  	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [netty-transport-4.1.76.Final.jar:?]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.76.Final.jar:?]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.76.Final.jar:?]
»  	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [netty-transport-4.1.76.Final.jar:?]
»  	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) [netty-transport-4.1.76.Final.jar:?]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722) [netty-transport-4.1.76.Final.jar:?]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:623) [netty-transport-4.1.76.Final.jar:?]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:586) [netty-transport-4.1.76.Final.jar:?]
»  	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496) [netty-transport-4.1.76.Final.jar:?]
»  	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) [netty-common-4.1.76.Final.jar:?]
»  	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.76.Final.jar:?]
»  	at java.lang.Thread.run(Thread.java:833) [?:?]
»  Caused by: javax.net.ssl.SSLException: org.bouncycastle.tls.TlsFatalAlert: handshake_failure(40)
»  	at org.bouncycastle.jsse.provider.ProvSSLEngine.unwrap(ProvSSLEngine.java:494) ~[bctls-fips-1.0.9.jar:?]
»  	at javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:679) ~[?:?]
»  	at io.netty.handler.ssl.SslHandler$SslEngineType$3.unwrap(SslHandler.java:295) ~[netty-handler-4.1.76.Final.jar:?]
»  	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1342) ~[netty-handler-4.1.76.Final.jar:?]
»  	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1235) ~[netty-handler-4.1.76.Final.jar:?]
»  	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1284) ~[netty-handler-4.1.76.Final.jar:?]
»  	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:510) ~[netty-codec-4.1.76.Final.jar:?]
»  	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:449) ~[netty-codec-4.1.76.Final.jar:?]
»  	... 16 more
»  Caused by: org.bouncycastle.tls.TlsFatalAlert: handshake_failure(40)
»  	at org.bouncycastle.tls.TlsProtocol.handleAlertWarningMessage(TlsProtocol.java:184) ~[bctls-fips-1.0.9.jar:?]
»  	at org.bouncycastle.tls.TlsServerProtocol.handleAlertWarningMessage(TlsServerProtocol.java:413) ~[bctls-fips-1.0.9.jar:?]
»  	at org.bouncycastle.tls.TlsProtocol.handleAlertMessage(TlsProtocol.java:161) ~[bctls-fips-1.0.9.jar:?]
»  	at org.bouncycastle.tls.TlsProtocol.processAlertQueue(TlsProtocol.java:570) ~[bctls-fips-1.0.9.jar:?]
»  	at org.bouncycastle.tls.TlsProtocol.processRecord(TlsProtocol.java:435) ~[bctls-fips-1.0.9.jar:?]
»  	at org.bouncycastle.tls.RecordStream.readFullRecord(RecordStream.java:184) ~[bctls-fips-1.0.9.jar:?]
»  	at org.bouncycastle.tls.TlsProtocol.safeReadFullRecord(TlsProtocol.java:727) ~[bctls-fips-1.0.9.jar:?]
»  	at org.bouncycastle.tls.TlsProtocol.offerInput(TlsProtocol.java:1059) ~[bctls-fips-1.0.9.jar:?]
»  	at org.bouncycastle.tls.TlsProtocol.offerInput(TlsProtocol.java:1027) ~[bctls-fips-1.0.9.jar:?]
»  	at org.bouncycastle.jsse.provider.ProvSSLEngine.unwrap(ProvSSLEngine.java:445) ~[bctls-fips-1.0.9.jar:?]
»  	at javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:679) ~[?:?]
»  	at io.netty.handler.ssl.SslHandler$SslEngineType$3.unwrap(SslHandler.java:295) ~[netty-handler-4.1.76.Final.jar:?]
»  	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1342) ~[netty-handler-4.1.76.Final.jar:?]
»  	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1235) ~[netty-handler-4.1.76.Final.jar:?]
»  	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1284) ~[netty-handler-4.1.76.Final.jar:?]
»  	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:510) ~[netty-codec-4.1.76.Final.jar:?]
»  	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:449) ~[netty-codec-4.1.76.Final.jar:?]
»  	... 16 more

```

**Comment by jakelandis** [comment]
@elastic/machine-learning - can you take a look at this test failure ? There are some TLS failures in the mix, but not sure if that is the root cause.

**Comment by pugnascotia** [comment]
Complete speculation - if we're using RAM disks, and I believe we are, and filled them up, would that trip us up? e.g.. ridiculous log spamming from something being completely broken?

**Comment by droberts195** [comment]
These are the sizes of the log directories that got uploaded to GCP at the end of the build:

```
bash-3.2$ du -sh *
116K    client
1.2M    daemon-2938.out.log
1.0M    distribution
9.7M    docs
464K    libs
 37M    modules
 14M    plugins
 30M    qa
 13M    rest-api-spec
301M    server
1.6M    test
383M    x-pack
```

So not tiny but not big enough to explain where 100GB of RAM went.

**Comment by mark-vieira** [comment]
So we actually [addressed this issue](https://github.com/elastic/elasticsearch/blob/1ba783ba05602f4c5533030bea55f54cde79223e/.ci/jobs.t/elastic%2Belasticsearch%2Bperiodic%2Bjava-matrix.yml#L22) for the normal java compatibility jobs, but not the FIPS ones.

Essentially the size of the project has grown where we cannot do a full `./gradlew check` in a ramdisk. We just produce too much output. The "solution" is to break this coverage into multiple builds, which we did elsewhere, but not in the FIPS testing case. I'll get this fixed.

**Comment by mark-vieira** [comment]
This should be fixed by 73da3249ff8657af11ccda7af71266d463db8c97.

**Comment by mark-vieira** [comment]
> So not tiny but not big enough to explain where 100GB of RAM went.

To clarify, we only upload a tiny fraction of the entire build workspace to GCP. Primarily logs, test cluster configuration, and possibly heap dumps if one was created. But the entire build checkout is done in ram, all build output (class files, jars, reports, etc) is also written to ram and that stuff can be quite large. Test output, specifically, is somewhat inefficiently duplicated a few times (binary format, junit-xml, html) as well.
